controllers/snapshot/snapshot_adapter.go

```diff
@@ -234,7 +245,18 @@ func (a *Adapter) EnsureStaticIntegrationPipelineRunsExist() (controller.Operati
 			} else {
 				pipelineRun, err := a.createIntegrationPipelineRun(a.application, &integrationTestScenario, a.snapshot)
 				if err != nil {
-					return a.HandlePipelineCreationError(err, &integrationTestScenario, testStatuses)
+					if clienterrors.IsInvalid(err) {
+						a.logger.Error(err, "pipelineRun failed during creation due to invalid resource",
+							"integrationScenario.Name", integrationTestScenario.Name)
+						testStatuses.UpdateTestStatusIfChanged(
+							integrationTestScenario.Name, intgteststat.IntegrationTestStatusTestInvalid,
+							fmt.Sprintf("Creation of pipelineRun failed during creation due to invalid resource: %s.", err))
+					} else {
+						a.logger.Error(err, "Failed to create pipelineRun for snapshot and scenario",
+							"integrationScenario.Name", integrationTestScenario.Name)
+					}
+					errsForPLRCreation = errors.Join(errsForPLRCreation, err)
```

Comment:
What if we do the `Join` only for errors that are not `Invalid`? 
So at the end ([here currently](https://github.com/redhat-appstudio/integration-service/pull/653/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R278)) we'd only need check if the `errsForPLRCreation` is not nil and then immediately `RequeueWithError` because we'll know there are no invalid errors caught there.

Reply:
Thanks. Updated.

Reply:
Thanks, this approach looks good to me.